### DISCUSSION TITLE
SST golang example

### DISCUSSION THREAD

**Discussion by gesaleh** [deepgram/community#942] [General help] [answered]
I'm trying to have the a working example for SST 
I use websocket, 

```
// ProcessAudioWithDeepgramSTT processes the audio using Deepgram's WebSocket API for STT and returns the transcription.
func ProcessAudioWithDeepgramSTT(clientSTT *deepgramstt.Client, audioStream []byte) (string, error) {
	// Connect to the Deepgram STT WebSocket
	bConnected := clientSTT.Connect()
	if !bConnected {
		fmt.Println("Client.Connect failed")
		return "", fmt.Errorf("error starting Deepgram STT WebSocket")
	}

	fmt.Println("Connected to Deepgram STT WebSocket. Processing audio...")

	// Create a channel to receive transcription updates

	go func() {
		// Feed the HTTP stream to the Deepgram client
		reader := bufio.NewReader(bytes.NewReader(audioStream))
		err := clientSTT.Stream(reader)
		if err != nil {
			fmt.Printf("Error streaming audio: %v", err)
			//return nil
		}
	}()

	return "", nil
}

```
I need the function to process the transcription received since this is a thread i need to have function to read the stream back i saw from api there are callback implementation but it is not clear from all the examples which one to use since API seems different from examples

**Comment by jpvajda** [comment, marked as answer]
@gesaleh  have you looked at our [Go WS Examples in our Go SDK for assistance?](https://github.com/deepgram/deepgram-go-sdk/tree/main/examples/speech-to-text/websocket)

These will all work with our SDK if you want to test them out. They are a good starting point, also you can look at our Go Starter apps as well

https://github.com/deepgram-starters/go-live-transcription
